### PR TITLE
WIP: Upgrade to 8.2

### LINKED ISSUE #12: Upgrade to 8.2
The list of packages has been compiled with `git diff 8.1..8.2.rc4 build/`.

**TODO**: Update this list with any SPKG changes that happen from 8.2.rc4 to 8.2.

🤔: These packages are probably not the right version in conda-forge but they should not matter for the conda release.

### *Base* SPKGs that have changed.
none.

### *Standard* SPKGs that have changed.
* [x] [alabaster](https://github.com/conda-forge/alabaster-feedstock/blob/master/recipe/meta.yaml) (0.7.8.p0 → 0.7.10), no patch changes
* [x] [arb](https://github.com/conda-forge/arb-feedstock/blob/master/recipe/meta.yaml) (2.11.1.p0 → 2.12.0.p0), no patch changes; https://github.com/conda-forge/arb-feedstock/pull/7
* [x] [babel](https://github.com/conda-forge/babel-feedstock/blob/master/recipe/meta.yaml) (2.3.4.p0 → 2.5.1), no patch changes
* [x] [BRiAl](https://github.com/conda-forge/brial-feedstock/blob/master/recipe/meta.yaml) (1.0.1.p1 → 1.0.1.p2), upgrades Sage's own build system
* [x] [backports.functools.lru_cache](https://github.com/conda-forge/backports.functools_lru_cache-feedstock/blob/master/recipe/meta.yaml) (× → 1.4)
* [x] [bleach](https://github.com/conda-forge/bleach-feedstock/blob/master/recipe/meta.yaml) (× → 2.1.3)
* [x] bzip2 (1.0.6-20150304 → 1.0.6-20150304.p0)
* [x] cddlib (094g.p1 → 094g.p2), build script changes
* [x] [certifi](https://github.com/conda-forge/certifi-feedstock/blob/master/recipe/meta.yaml) (2016.2.28.p0 → 2017.11.5)
* [x] [cliquer](https://github.com/conda-forge/cliquer-feedstock/blob/master/recipe/meta.yaml) (1.21.p3 → 1.21.p4), build scirpt changes
* [x] configure (247 → 261)
* [x] [cypari](https://github.com/conda-forge/cypari2-feedstock/blob/master/recipe/meta.yaml) (1.1.3 → 1.1.4)
* [x] [cysignals](https://github.com/conda-forge/cysignals-feedstock/blob/master/recipe/meta.yaml) ([1.6.5 → 1.7.0.p0](https://github.com/conda-forge/cysignals-feedstock/pull/11)), added `stack_guard.patch`
* [x] [cython](https://github.com/conda-forge/cython-feedstock/blob/master/recipe/meta.yaml) (0.27.2 → 0.28.1)
* [x] [docutils](https://github.com/conda-forge/docutils-feedstock/blob/master/recipe/meta.yaml) (0.13.1 → 0.14)
* [x] ecl (16.1.2.p4 → 16.1.2.p5), only changes to build system
* [x] eclib (20171002 → 20171002.p0), only changes to build system
* [x] ecm (7.0.4.p0 → 7.0.4.p1), only changes to build system
* [x] [entrypoints](https://github.com/conda-forge/entrypoints-feedstock/blob/master/recipe/meta.yaml) (0.2.2 → 0.2.3)
* [x] [fflas_ffpack](https://github.com/conda-forge/fflas-ffpack-feedstock/blob/master/recipe/meta.yaml) ([2.2.2.p0 → 2.3.2](https://github.com/conda-forge/fflas-ffpack-feedstock/pull/5)), drops `ticket-21578.patch`
* [x] flint (2.5.2.p1 → 2.5.2.p2), patch for FreeBSD added.
* [x] flintqs (1.0 → 1.0.p0), only build system changes
* [x] fplll (5.2.0 → 5.2.0.p1), patch for Solaris (and nobody else I hope)
* [x] [freetype](https://github.com/conda-forge/freetype-feedstock/blob/master/recipe/meta.yaml) (2.6.3 → 2.8.1)
* [x] [future](https://github.com/conda-forge/future-feedstock/blob/master/recipe/meta.yaml) (0.15.2.p0 → 0.16.0)
* [x] [gc](https://github.com/conda-forge/bdw-gc-feedstock/blob/master/recipe/meta.yaml) (7.2f.p1 → 7.6.4)
* [x] gf2x (1.2 → 1.2.p0), only build system changes
* [x] [giac](https://github.com/conda-forge/giac-feedstock/blob/master/recipe/meta.yaml) (1.2.3.47.p0 → 1.4.9.45.p2), drops `cSolveorder-check.patch`, `libpng16.patch`, adds `dirent.patch`, `alloca.patch`, `sunos_abs.patch`
* [x] [givaro](https://github.com/conda-forge/givaro-feedstock/blob/master/recipe/meta.yaml) ([4.0.2.p0 → 4.0.4](https://github.com/conda-forge/givaro-feedstock/pull/4))
* [x] glpk (4.63.p1 → 4.63.p2), just build system changes
* [x] gsl (2.3 → 2.3.p0), just build system changes
* [x] [html5lib](https://github.com/conda-forge/html5lib-feedstock/blob/master/recipe/meta.yaml) (× → 1.0.1)
* [x] [iconv](https://github.com/conda-forge/libiconv-feedstock/blob/master/recipe/meta.yaml) (1.14 → 1.15)
* [x] iml (1.0.4p1.p1 → 1.0.4p1.p2), just build system changes
* [x] [ipykernel](https://github.com/conda-forge/ipykernel-feedstock/blob/master/recipe/meta.yaml) (4.6.1 → 4.7.0)
* [x] 🤔 [ipython](https://github.com/conda-forge/ipython-feedstock/blob/master/recipe/meta.yaml) (5.1.0 → 5.5.0), conda packages version 6, not 5.
* [x] [ipywidgets](https://github.com/conda-forge/ipywidgets-feedstock/blob/master/recipe/meta.yaml) (6.0.0 → 7.1.1)
* [x] [jinja2](https://github.com/conda-forge/jinja2-feedstock/blob/master/recipe/meta.yaml) (2.8.p0 → 2.10)
* [x] 🤔 jmol (14.6.1_2016.07.11.p0 → 14.6.1_2016.07.11.p1), we did not have that one packaged before either
* [x] [jsonschema](https://github.com/conda-forge/jsonschema-feedstock/blob/master/recipe/meta.yaml) (2.5.1.p0 → 2.6.0)
* [x] [jupyter_client](https://github.com/conda-forge/jupyter_client-feedstock/blob/master/recipe/meta.yaml) (5.1.0 → 5.2.2)
* [x] [jupyter_core](https://github.com/conda-forge/jupyter_core-feedstock/blob/master/recipe/meta.yaml) (4.3.0 → 4.4.0)
* [x] [lcalc](https://github.com/conda-forge/lcalc-feedstock/blob/master/recipe/meta.yaml) (1.23.p15 → 1.23.p16), adds `pari-mem.patch`, seems to be only relevant for cygwin.
* [x] [libatomic_ops](https://github.com/conda-forge/libatomic_ops-feedstock/blob/master/recipe/meta.yaml) ([× → 7.6.2](https://github.com/conda-forge/libatomic_ops-feedstock/pull/11))
* [x] libgap (4.8.6 → 4.8.6.p0), only build system changes
* [x] libgd (2.1.1.1.p0 → 2.1.1.1.p1), only build system changes
* [x] libpng (1.6.29 → 1.6.29.p0), only build system changes
* [x] [linbox](https://github.com/conda-forge/linbox-feedstock/blob/master/recipe/meta.yaml) ([1.4.2.p1 → 1.5.2](https://github.com/conda-forge/linbox-feedstock/pull/5)), lots of patch changes
* [x] lrcalc (1.2.p0 → 1.2.p1), only build system changes
* [x] m4ri (20140914 → 20140914.p0), only build system changes
* [x] m4rie (20150908 → 20150908.p0), only build system changes
* [x] [markupsafe](https://github.com/conda-forge/markupsafe-feedstock/blob/master/recipe/meta.yaml) (0.23.p0 → 1.0)
* [x] [mpfi](https://github.com/conda-forge/mpfi-feedstock/blob/master/recipe/meta.yaml) ([1.5.1 → 1.5.2](https://github.com/conda-forge/mpfi-feedstock/pull/2))
* [x] [mpc](https://github.com/conda-forge/mpc-feedstock/blob/master/recipe/meta.yaml) ([1.0.3.p0 → 1.1.0](https://github.com/conda-forge/mpc-feedstock/pull/12)), drop `mpc_mul_faster.patch`
* [x] [mpfr](https://github.com/conda-forge/mpfr-feedstock/blob/master/recipe/meta.yaml) ([3.1.5.p0 → 4.0.1](https://github.com/conda-forge/mpfr-feedstock/pull/17)), drop `clang_workaround.patch`
* [x] [mpmath](https://github.com/conda-forge/mpmath-feedstock/blob/master/recipe/meta.yaml) (0.19.p0 → 1.0.0)
* [x] 🤔 nbconvert (4.2.0.p0 → 5.3.1)
* [x] [ncurses](https://github.com/conda-forge/ncurses-feedstock/blob/master/recipe/meta.yaml) (6.0 → 6.0.p0)
* [x] 🤔 notebook (4.4.1.p0 → 5.3.1), we had not packaged the notebook before and sage worked without.
* [x] numpy (1.13.3.p0 → 1.13.3.p1), build system changes
* [x] [pandocfilters](https://github.com/conda-forge/pandocfilters-feedstock/blob/master/recipe/meta.yaml) ([1.3.0 → 1.4.2](https://github.com/conda-forge/pandocfilters-feedstock/pull/3)), upgraded to standard
* [x] [pathlib2](https://github.com/conda-forge/pathlib2-feedstock/blob/master/recipe/meta.yaml) (2.1.0.p0 → 2.3.0)
* [x] pcre (8.40.p0 → 8.40.p2), build system changes only
* [x] [pickleshare](https://github.com/conda-forge/pickleshare-feedstock/blob/master/recipe/meta.yaml) (0.7.2.p0 → 0.7.4)
* [x] [pillow](https://github.com/conda-forge/pillow-feedstock/blob/master/recipe/meta.yaml) (3.3.0 → 4.3.0), currently conda is on a different major version but 4.3.0 was packaged here https://github.com/conda-forge/pillow-feedstock/commit/e93af418d51082fc2021733072388b51894e9013#diff-e178b687b10a71a3348107ae3154e44c
* [x] [pip](https://github.com/conda-forge/pip-feedstock/blob/master/recipe/meta.yaml) (9.0.1 → 9.0.3)
* [x] pkgconf (0.9.7 → 0.9.7.p0), build system changes only
* [x] planarity (3.0.0.5 → 3.0.0.5.p0), build system changes only
* [x] [ppl](https://github.com/conda-forge/ppl-feedstock/blob/master/recipe/meta.yaml) (1.2 → 1.2.p1), added `clang5-support.patch`
* [x] [prompt_toolkit](https://github.com/conda-forge/prompt_toolkit-feedstock/blob/master/recipe/meta.yaml) (1.0.9 → 1.0.15)
* [x] [psutil](https://github.com/conda-forge/psutil-feedstock/blob/master/recipe/meta.yaml) (5.2.0.p0 → 5.2.0.p1), only changes to cygwin patches
* [x] [pynac](https://github.com/conda-forge/pynac-feedstock/blob/master/recipe/meta.yaml) (0.7.12 → 0.7.16.p0), added `memleak.patch`
* [x] [pyparsing](https://github.com/conda-forge/pyparsing-feedstock/blob/master/recipe/meta.yaml) (2.1.5.p0 → 2.2.0)
* [x] [pytz](https://github.com/conda-forge/pytz-feedstock/blob/master/recipe/meta.yaml) (2016.4.p0 → 2017.3)
* [x] 🤔 [pyzmq](https://github.com/conda-forge/pyzmq-feedstock/blob/master/recipe/meta.yaml) (16.0.0 → 17.0.0b3)
* [x] 🤔 r (3.4.2.p0 → 3.4.3.p0), we still have 3.4.1 in conda, let's see if that is good enough
* [x] [readline](https://github.com/conda-forge/readline-feedstock/blob/master/recipe/meta.yaml) (6.3.008 → 6.3.008.p0), conda has 7.0
* [x] [rw](https://github.com/conda-forge/rw-feedstock/blob/master/recipe/meta.yaml) (0.7 → 0.7.p0)
* [x] [scandir](https://github.com/conda-forge/scandir-feedstock/blob/master/recipe/meta.yaml) (× → 1.6)
* [x] [send2trash](https://github.com/conda-forge/send2trash-feedstock/blob/master/recipe/meta.yaml) (× → 1.4.2)
* [x] [setuptools](https://github.com/conda-forge/setuptools-feedstock/blob/master/recipe/meta.yaml) (33.1.1 → 38.2.4.p0), build system changes
* [x] [setuptools_scm](https://github.com/conda-forge/setuptools_scm-feedstock/blob/master/recipe/meta.yaml) (1.11.1.p0 → 1.15.6)
* [x] 🤔 [singular](https://github.com/conda-forge/singular-feedstock/blob/master/recipe/meta.yaml) (4.1.0p3.p0 → 4.1.0p3.p2), adds a few patches
* [x] [sphinxcontrib_websupport](https://github.com/conda-forge/sphinxcontrib-websupport-feedstock/blob/master/recipe/meta.yaml) (× → 1.0.1)
* [x] [sqlite](https://github.com/conda-forge/sqlite-feedstock/blob/master/recipe/meta.yaml) (autoconf-3170000 → 3220000)
* [x] [subprocess32](https://github.com/conda-forge/subprocess32-feedstock/blob/master/recipe/meta.yaml) (× → 3.2.7.p0), adds `fix_config.py` which seems to be relevant for Solaris only
* [x] 🤔 [sympy](https://github.com/conda-forge/sympy-feedstock/blob/master/recipe/meta.yaml) ([1.1.1 → 1.1.1.p0](https://github.com/conda-forge/sympy-feedstock/pull/9)), adds `prec-to-precision-in-core-numbers.patch`
* [x] [terminado](https://github.com/conda-forge/terminado-feedstock/blob/master/recipe/meta.yaml) (0.6 → 0.8.1)
* [x] [testpath](https://github.com/conda-forge/testpath-feedstock/blob/master/recipe/meta.yaml) (× → 0.3.1)
* [x] [tornado](https://github.com/conda-forge/tornado-feedstock/blob/master/recipe/meta.yaml) (4.3.p0 → 4.5.2), conda has version 5 but that should be fine
* [x] [traitlets](https://github.com/conda-forge/traitlets-feedstock/blob/master/recipe/meta.yaml) (4.3.1 → 4.3.2)
* [x] [typing](https://github.com/conda-forge/typing-feedstock/blob/master/recipe/meta.yaml) (3.6.1 → 3.6.2)
* [x] [webencodings](https://github.com/conda-forge/webencodings-feedstock/blob/master/recipe/meta.yaml) ([× → 0.5.1](https://github.com/conda-forge/webencodings-feedstock/pull/2))
* [x] [widgetsnbextension](https://github.com/conda-forge/widgetsnbextension-feedstock/blob/master/recipe/meta.yaml) (2.0.0 → 3.1.1)
* [x] [xz](https://github.com/conda-forge/xz-feedstock/blob/master/recipe/meta.yaml) (5.2.2 → 5.2.2.p0)
* [x] [yasm](https://github.com/conda-forge/yasm-feedstock/blob/master/recipe/meta.yaml) (1.3.0 → 1.3.0.p0), build system changes only
* [x] zeromq (4.0.5.p1 → 4.0.5.p2), build system changes only
* [x] [zope_interface](https://anaconda.org/anaconda/zope.interface) (4.2.0.p0 → 4.4.3), where is this package in conda-forge?

### *Optional* SPKGs that have changed.
😞: Proper support for optional packages won't come before Sage 8.3 with the inclusion of https://trac.sagemath.org/ticket/20382. These packages are probably not present in conda-forge but it does not matter at the moment as they would not be detected by Sage anyway.
* [x] [ccache](https://github.com/conda-forge/ccache-feedstock/blob/master/recipe/meta.yaml) (3.2.2 → 3.3.4)
* [x] coxeter3 (1.1 → 8ac9c71723c8ca57a836d6381aed125261e44e9e)
* [x] gambit (15.1.1 → 15.1.1.p0)
* [x] gmpy2 (2.0.8.p0 → 2.1.0a1.p1)
* [x] libhomfly (1.0r2 → 1.02r4)
* [x] meataxe (2.4.24.p3 → 1.0)
* [x] [matplotlib](https://github.com/conda-forge/matplotlib-feedstock/blob/master/recipe/meta.yaml) (1.5.1.p0 → 2.1.0)
* [x] maxima (5.39.0.p0 → 5.39.0.p2), only build system changes
* [x] [mistune](https://github.com/conda-forge/mistune-feedstock/blob/master/recipe/meta.yaml) (0.7.3.p0 → 0.8.3)
* [x] notedown (dfda648b → 1.5.1)
* [x] openssl (1.0.2m → 1.1.0h)
* [x] pari_jupyter (1.2.2 → 1.3.2)
* [x] qhull (2015-src-7.2.0 → 2015-src-7.2.0.p1)
* [x] singular_jupyter (0.9.2 → 0.9.7)

### *Experimental* SPKGs that have changed.
😞: Similarly as for optional SPKGs we don't really care yet.
none.

### New runtime dependencies:
The dependencies of the `sagelib` target have not changed upstream, so all new packages should be runtime dependencies in the `sage` package.
* [x] [backports.functools.lru_cache](https://github.com/conda-forge/backports.functools_lru_cache-feedstock/blob/master/recipe/meta.yaml) (× → 1.4)
* [x] [libatomic_ops](https://github.com/conda-forge/libatomic_ops-feedstock/blob/master/recipe/meta.yaml) (× → 7.6.2)
* [x] [scandir](https://github.com/conda-forge/scandir-feedstock/blob/master/recipe/meta.yaml) (× → 1.6)
* [x] [send2trash](https://github.com/conda-forge/send2trash-feedstock/blob/master/recipe/meta.yaml) (× → 1.4.2)
* [x] [sphinxcontrib_websupport](https://github.com/conda-forge/sphinxcontrib-websupport-feedstock/blob/master/recipe/meta.yaml) (× → 1.0.1)
* [x] [subprocess32](https://github.com/conda-forge/subprocess32-feedstock/blob/master/recipe/meta.yaml) (× → 3.2.7.p0)

### New build-time dependencies:
The dependencies of the `sagelib` target have not changed upstream, so there should be no new build-time dependencies.

### PR DESCRIPTION
**Pull request by saraedum**
Fixes #12.

### CODE REVIEW

**Comment by saraedum** [comment]
The sagelib upgrade is here: https://github.com/conda-forge/sagelib-feedstock/pull/22

**Comment by saraedum** [comment]
@conda-forge-admin please rerender

**Comment by saraedum** [comment]
@conda-forge-admin please rerender

**Comment by saraedum** [comment]
Pending on https://github.com/conda-forge/maxima-feedstock/pull/7

File: recipe/meta.yaml [hunk 1]
```diff
@@ -1,117 +1,123 @@
...
-    - fflas-ffpack >=2.2.2,<2.3.0
+    - entrypoints >=0.2.3
+    - fflas-ffpack >=2.3.2,<2.4.0
     - flintqs >=1.0,<2
     - fplll >=5.2.0,<6
     - fpylll >=0.3.0dev
-    - freetype >=2.6.3,<3
+    # currently, conda-forge's matplotlib builds against freetype=2.6, so we
+    # can not use the version 2.8 that SageMath proposes.
+    - freetype >=2.6,<3
```
Comment (isuruf) [inline recipe/meta.yaml]: Instead of downgrading freetype, can you upgrade the matplotlib version?

Comment (saraedum) [reply recipe/meta.yaml]: Sure. Let's do this as well.

Comment (saraedum) [reply recipe/meta.yaml]: Actually the latest matplotlib does use 2.8.*. Let's see why sage wanted to pull in a later version.

Comment (saraedum) [reply recipe/meta.yaml]: Oh, now I understood what you meant. Upgrade matplotlib here. I thought you meant upgrade matplotlib so it uses the latest freetype.

**Comment by saraedum** [comment]
Awesome, thanks for your help @isuruf!

**Pull request merged**